package com.zbkj.common.constants;

/**
 * 用户常量表
 *  +----------------------------------------------------------------------
 *  | CRMEB [ CRMEB赋能开发者，助力企业发展 ]
 *  +----------------------------------------------------------------------
 *  | Copyright (c) 2016~2025 https://www.crmeb.com All rights reserved.
 *  +----------------------------------------------------------------------
 *  | Licensed CRMEB并不是自由软件，未经许可不能去掉CRMEB相关版权
 *  +----------------------------------------------------------------------
 *  | Author: CRMEB Team <admin@crmeb.com>
 *  +----------------------------------------------------------------------
 */
public class UserConstants {

    /** 用户登录方式 Google */
    public static final String USER_LOGIN_TYPE_GOOGLE = "google";
    /** 用户登录方式 邮箱 */
    public static final String USER_LOGIN_TYPE_EMAIL = "email";
    /** 用户登录方式 Facebook */
    public static final String USER_LOGIN_TYPE_FACEBOOK = "facebook";
    /** 用户登录方式 手机号 */
    public static final String USER_LOGIN_TYPE_PHONE = "phone";
    /** 用户登录方式 twitter */
    public static final String USER_LOGIN_TYPE_TWITTER = "twitter";
    /** 用户登录方式 游客 */
    public static final String USER_LOGIN_TYPE_VISITOR = "visitor";

    /**
     * =========================================================
     * 用户搜索部分
     * =========================================================
     */

    /** 用户搜索类型——all */
    public static final String USER_SEARCH_TYPE_ALL = "all";
    /** 用户搜索类型——UID 精准匹配 */
    public static final String USER_SEARCH_TYPE_UID = "uid";
    /** 用户搜索类型——用户昵称 全模糊匹配 */
    public static final String USER_SEARCH_TYPE_NICKNAME = "nickname";
    /** 用户搜索类型——手机号 精准匹配 */
    public static final String USER_SEARCH_TYPE_PHONE = "phone";
}
